packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/utils.ts

```diff
@@ -0,0 +1,127 @@
+import { UniqueIdentifier } from "@dnd-kit/core";
+import { arrayMove } from "@dnd-kit/sortable";
+import { groupBy } from "lodash";
+import { AccountPage } from "../../../../components/hooks/useAccountPages";
+
+export interface PageList {
+  treeList: TreeElement[];
+  flatList: TreeElement[];
+}
+
+export interface TreeElement extends AccountPage {
+  parentId: string;
+  depth: number;
+  index: number;
+  expanded: boolean;
+  expandable: boolean;
+  collapsed: boolean;
+  children?: TreeElement[];
+}
+
+const getRecursivePageList = (
+  groupedPages: { [id: string]: AccountPage[] },
+  id: string,
+  expandedIds: string[],
+  depth = 0,
+  collapsed = false,
+): PageList => {
+  const emptyTreeList = [] as TreeElement[];
+  const emptyList: PageList = {
+    treeList: emptyTreeList,
+    flatList: emptyTreeList,
+  };
+
+  return (
+    groupedPages[id]?.reduce((prev, page, index) => {
+      const expanded = expandedIds.includes(page.entityId);
+      const children = getRecursivePageList(
+        groupedPages,
+        page.entityId,
+        expandedIds,
+        depth + 1,
+        collapsed || !expanded,
+      );
+      const expandable = !!children.treeList.length;
+
+      const item = {
+        ...page,
+        depth,
+        index,
+        parentId: page.parentPageEntityId,
+        expanded,
+        expandable,
+        collapsed,
+      } as TreeElement;
+
+      return {
+        treeList: [
+          ...prev.treeList,
+          {
+            ...item,
+            children: children.treeList,
+          } as TreeElement,
+        ],
+        flatList: [...prev.flatList, item, ...children.flatList],
+      };
+    }, emptyList) || emptyList
+  );
+};
+
+export const getPageList = (pages: AccountPage[], expandedIds: string[]) =>
+  getRecursivePageList(
+    groupBy(pages, (page) => page.parentPageEntityId),
+    "null",
+    expandedIds,
+    0,
+  );
+
+export const getProjection = (
+  items: TreeElement[],
+  activeId: UniqueIdentifier,
+  overId: UniqueIdentifier,
+  dragOffset: number,
+  indentationWidth: number,
+) => {
+  const filteredItems = items.filter((item) => !item.collapsed);
+  const overItemIndex = filteredItems.findIndex(
+    ({ entityId }) => entityId === overId,
+  );
+  const activeItemIndex = filteredItems.findIndex(
+    ({ entityId }) => entityId === activeId,
+  );
+
+  const activeItem = filteredItems[activeItemIndex];
+  const newItems = arrayMove(filteredItems, activeItemIndex, overItemIndex);
+  const previousItem = newItems[overItemIndex - 1];
+  const nextItem = newItems[overItemIndex + 1];
+
+  const dragDepth = Math.round(dragOffset / indentationWidth);
```

Comment:
I think `dragDepth` should be the argument here, not `dragOffset` and `indentationWidth`, which are two variables not relevant to this function beyond calculating `dragDepth`.

Reply:
Moved dragDepth calculation outside of this function in c762b94d0bdc79fd988f6ed3b7364ed3f31e24a9